src/components/marker-table/index.js

```diff
@@ -148,6 +97,72 @@ class MarkerTree {
   }
 }
 
+export function getMarkerName(marker: Marker) {
+  let name = marker.name;
+
+  if (marker.data) {
+    const data = marker.data;
+    switch (data.type) {
+      case 'tracing':
+        if (typeof data.category === 'string') {
+          if (data.category === 'log' && name.length > 100) {
+            name = name.substring(0, 100) + '...';
+          } else if (data.category === 'DOMEvent') {
+            name = data.eventType;
+          }
+        }
+        break;
+      case 'UserTiming':
+        name = data.name;
+        break;
+      case 'FileIO':
+        if (data.source) {
+          name = `(${data.source}) `;
+        }
+        name += data.operation;
+        if (data.filename) {
+          name = data.operation ? `${name} — ${data.filename}` : data.filename;
+        }
+        break;
+      case 'Text':
+        name += ` — ${data.name}`;
+        break;
+      default:
+    }
+  }
+  return name;
+}
```

Comment:
Thanks a lot for creating these functions.
Can you please move this one to `src/profile-logic/marker-data`; and also call it `getMarkerFullDescription` to better reflect what it does.

Reply:
@julienw, I just committed these changes, take another look.